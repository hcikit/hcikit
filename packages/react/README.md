# @hcikit/react

This package is a package that allows you to create and run HCI experiments. Visit hcikit.github.io for more information.
