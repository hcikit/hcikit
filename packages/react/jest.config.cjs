/** @type {import('@ts-jest/dist/types').InitialOptionsTsJest} */
module.exports = {
  preset: "ts-jest",
  testEnvironment: "jsdom",
  moduleNameMapper: {
    "@hcikit/workflow(.*)$": "<rootDir>/../workflow/src$1",
  },
  testMatch: ["./src/**/*.[jt]s?(x)", "**/?(*.)+(spec|test).[t]s?(x)"],
  modulePathIgnorePatterns: ["./dist/", "./node_modules/"],
  testPathIgnorePatterns: ["./dist/", "./node_modules/"],
  transformIgnorePatterns: ["./dist/", "/node_modules/"],
  setupFilesAfterEnv: ["<rootDir>/setupTests.ts"],
  collectCoverageFrom: [
    "src/**/*.{ts,tsx}",
    "!/node_modules/",
    "!src/setupTests.ts",
    "!src/**/*.stories.{js,ts,tsx,jsx}",
  ],
};
