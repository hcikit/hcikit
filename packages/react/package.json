{
  "name": "@hcikit/react",
  "version": "1.0.0-alpha.14",
  "description": "An experiment runner with a set of pre-built experiment components for running hci experiments",
  "author": "blainelewis1",
  "license": "MIT",
  "repository": "hcikit/hcikit",
  "module": "dist/index.js",
  "main": "dist/index.js",
  "type": "module",
  "keywords": [
    "react",
    "hci",
    "experiment"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    "import": "./dist/index.js",
    "default": "./dist/index.js"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "engines": {
    "node": ">=8",
    "npm": ">=5"
  },
  "scripts": {
    "test": "NODE_OPTIONS=--experimental-vm-modules yarn jest --env=jsdom",
    "test:watch": "NODE_OPTIONS=--experimental-vm-modules yarn jest --watch --env=jsdom --colors --verbose",
    "start": "tsc --watch",
    "build": "tsc --build --verbose",
    "prepack": "yarn run build",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "peerDependencies": {
    "@mui/icons-material": "^5.5.0",
    "@mui/material": "^5.5.0",
    "@mui/styles": "^5.5.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "dependencies": {
    "@emotion/react": "^11.11.1",
    "@emotion/styled": "^11.11.0",
    "@hcikit/workflow": "workspace:*",
    "core-js": "^3.32.2",
    "lodash-es": "^4.17.21",
    "prop-types": "^15.8.1",
    "react-error-boundary": "^4.0.11",
    "react-is": "^18.2.0",
    "react-markdown": "^8.0.7",
    "serialize-error": "^11.0.2"
  },
  "gitHead": "8d0f2af41121f70996e8b5803d006832b147fd86",
  "devDependencies": {
    "@emotion/react": "^11.11.1",
    "@emotion/styled": "^11.11.0",
    "@mui/icons-material": "^5.14.11",
    "@mui/material": "^5.14.11",
    "@mui/styles": "^5.14.11",
    "@testing-library/jest-dom": "^6.1.3",
    "@testing-library/react": "^14.0.0",
    "@types/jest": "^29.5.5",
    "@types/react": "^18.2.23",
    "jest-diff": "^29.7.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "^5.0.1",
    "typescript": "^5.2.2"
  }
}
