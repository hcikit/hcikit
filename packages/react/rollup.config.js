import babel from "rollup-plugin-babel";
import commonjs from "@rollup/plugin-commonjs";
import resolve from "@rollup/plugin-node-resolve";
import pkg from "./package.json";
import postcss from "rollup-plugin-postcss";

export default {
  input: "src/index.js",
  output: [
    {
      file: pkg.main,
      format: "cjs",
      sourcemap: true,
    },
    {
      file: pkg.module,
      format: "es",
      sourcemap: true,
    },
  ],
  external: Object.keys(pkg.peerDependencies || {}),
  plugins: [
    resolve(),
    commonjs({
      exclude: [
        "src/**",
        "../../node_modules/lodash-es/**",
        "../../node_modules/symbol-observable/es/**",
      ],
      include: ["../../node_modules/**"],
      namedExports: {
        "../../node_modules/react-redux/node_modules/react-is/index.js": [
          "isValidElementType",
          "isContextConsumer",
        ],
        "../../node_modules/@material-ui/utils/node_modules/react-is/index.js": [
          "ForwardRef",
          "Memo",
        ],
        "../../node_modules/react-dom/index.js": [
          "findDOMNode",
          "unstable_batchedUpdates",
        ],
        "../../node_modules/react-is/index.js": [
          "isValidElementType",
          "isElement",
          "ForwardRef",
          "typeOf",
          "Memo",
          "isContextConsumer",
        ],
        "../../node_modules/prop-types/index.js": ["element", "elementType"],
      },
    }),
    babel({
      exclude: "../../node_modules/**",
    }),
    postcss(),
  ],
};
