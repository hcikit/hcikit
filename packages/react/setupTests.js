import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let error = console.error;

console.error = function (message) {
  error.apply(console, arguments); // keep default behaviour
  throw message instanceof Error ? message : new Error(message);
};
