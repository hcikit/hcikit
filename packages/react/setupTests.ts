import "@testing-library/jest-dom";

console.log("heloo");

// jest.mock("styled-components", () => {
//   const actual = jest.requireActual(
//     "styled-components"
//   ) as typeof import("styled-components");
//   const styled = actual.default;

//   return Object.assign(styled, actual);
// });

// const error = console.error;

// console.error = function (message, ...args) {
//   error.apply(console, args); // keep default behaviour
//   throw message instanceof Error ? message : new Error(message);
// };
