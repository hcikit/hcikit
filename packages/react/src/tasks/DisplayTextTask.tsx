import React from "react";
import { Button } from "@mui/material";
import { CenteredDiv, CenteredText } from "../components/index.js";
import PropTypes from "prop-types";
import { withGridItem } from "../GridLayout.js";
import { useExperiment } from "../core/Experiment.js";
/**
 * Creates a simple ane small piece of text in the middle of the screen with a continue button.
 */
const DisplayTextTask: React.FunctionComponent<{ content: string }> = ({
  content,
}) => {
  let experiment = useExperiment();
  return (
    <CenteredDiv centerX centerY>
      <CenteredText>
        <h1>{content}</h1>
        <Button
          variant="contained"
          color="primary"
          onClick={() => experiment.advance()}
        >
          Continue
        </Button>
      </CenteredText>
    </CenteredDiv>
  );
};

DisplayTextTask.propTypes = {
  content: PropTypes.string.isRequired,
};

export default withGridItem(DisplayTextTask);
