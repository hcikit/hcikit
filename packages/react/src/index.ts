import { PropsWithChildren, ReactElement, Component } from "react";
import { ControlFunctions } from "./core/Experiment.js";
import * as allTasks from "./tasks/index.js";

export * from "./tasks/index.js";
export * from "./components/index.js";

export { allTasks };

export { withGridItem } from "./GridLayout.js";

export { default as GridLayout } from "./GridLayout.js";
export { default } from "./core/Experiment.js";
export * from "./core/Experiment.js";
export {
  BasePersistence,
  StoragePersistence,
  FileSystemPersistence,
} from "./persistence/index.js";

type TaskProps = ControlFunctions;
type StaticProps = {
  layoutProps?: Record<string, any>;
};

export type Task<P = {}> = React.ComponentType<TaskProps & P> & StaticProps;

export interface FunctionTask<P = {}>
  extends React.FunctionComponent<P & ControlFunctions> {
  (
    props: PropsWithChildren<P & ControlFunctions>,
    context?: any
  ): ReactElement<any, any> | null;
}

export abstract class ClassTask<P = {}, S = {}, SS = any> extends Component<
  P & TaskProps,
  S,
  SS
> {}
