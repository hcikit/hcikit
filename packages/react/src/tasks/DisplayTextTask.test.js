import React from "react";
import DisplayTextTask from "./DisplayTextTask";
import { screen, render } from "@testing-library/react";

describe("DisplayTextTask", () => {
  it("renders without crashing", () => {
    render(<DisplayTextTask content="hello" />);
  });

  // TODO: this will kind of be ruined.. Unless I mock the provider.
  it("advances to the next task", () => {
    let taskCompleteSpy = jest.fn();

    render(
      <DisplayTextTask taskComplete={taskCompleteSpy} content="Hello World" />
    );

    screen.getByText("Continue").click();
    expect(taskCompleteSpy).toBeCalledTimes(1);
  });

  it("renders properly", () => {
    expect(
      render(<DisplayTextTask content="hello" />).asFragment()
    ).toMatchSnapshot();
  });

  it("renders plain text", () => {
    render(<DisplayTextTask content="hello" />);

    screen.getByText("hello", { selector: "h1" });
  });
});
