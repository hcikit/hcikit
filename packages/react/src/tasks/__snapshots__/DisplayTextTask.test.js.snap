// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayTextTask renders properly 1`] = `
<div
  style={
    Object {
      "gridArea": "task",
    }
  }
>
  <div
    className="sc-AxhUy gYMEmy"
  >
    <div>
      <div
        className="sc-AxgMl ftqCsy"
      >
        <h1>
          hello
        </h1>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Continue
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
